OverlayPlugin.Updater/HttpClientWrapper.cs

```diff
@@ -50,28 +60,25 @@ public static string Get(string url)
                     else
                     {
                         var buffer = new byte[40 * 1024];
-                        var length = 0;
-
-                        IEnumerable<string> lengthValues;
-                        if (response.Headers.TryGetValues("Content-Length", out lengthValues))
-                        {
-                            int.TryParse(lengthValues.GetEnumerator().Current, out length);
-                        }
+                        var length = (long)response.Content.Headers.ContentLength;
```

Comment:
This will still throw the same exception even with the latest changes in commit 80a60c5. Consider the following setup:

---

A test web server that does not send a `Content-Length` header. I used a simple PHP script with the built-in PHP web server (`php -S 0.0.0.0:12345`):

```php
<?php

for ($i = 0; $i < 1024; ++$i) echo chr(rand(0x30, 0x7A));
```

---

A simple test executable with the same logic as this PR attempting to read the response:

```cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;
using System.Linq;
using System.Net.Http;
using System.Reflection;
using System.Threading;
using System.Threading.Tasks;

namespace TestHttpClient {
    class MainClass {
        public static void Main(string[] _) {
            var task = Task.Run(async () => {
                var client = new HttpClient();

                var request = new HttpRequestMessage()
                {
                    RequestUri = new Uri("http://127.0.0.1:12345/test.php"),
                    Method = HttpMethod.Get,
                };

                var response = await client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);

                var buffer = new byte[40 * 1024];
                long length = 0;

                IEnumerable<string> lengthValues;
                if (response.Headers.TryGetValues("Content-Length", out lengthValues))
                {
                    long.TryParse(lengthValues.GetEnumerator().Current, out length);
                }
                if (length == 0)
                {
                    length = (long)response.Content.Headers.ContentLength;
                }
                
                var result = await response.Content.ReadAsStringAsync();

                Console.WriteLine(result);
            });
            task.Wait();
        }
    }
}
```

(Compiled with `csc /R:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.8\System.Net.Http.dll" test.cs`)

---

Running as-is produces this exception:

```
PS C:\Users\valarnin\Source\test> .\test.exe

Unhandled Exception: System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: Nullable object must have a value.
   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   at System.Nullable`1.get_Value()
   at TestHttpClient.MainClass.<>c.<<Main>b__0_0>d.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at TestHttpClient.MainClass.Main(String[] _)
```

Commenting out line 35 of the test.cs file (`length = (long)response.Content.Headers.ContentLength;`) no longer triggers the exception:

```
PS C:\Users\valarnin\Source\test> .\test.exe
v?Frx=WphTVh^iAEje4v0?qU;@U]t_4spTo>DGge`rKRcU0ov?toWybU3B`bDhPz4;2@CpJZ53^TTj=\t_w@d8Jb8PaxK]PB@1SkUb[kwNhb74JVV93hi8I8?my4gkO2CZ;GvZLO]oDQn5<gUEbpMaR;O2P[0u3dGzFY=SN?s>1N1xKecSFx[[cRFdyygUF^ohA911b[tXtNqQGmdnisli6S5z;tre7MXyCur9757<icABmIzuza6L_]6TP3[zNzax?sT_AkzRj@SlBnzF_eZznzk_]lb5vvt=x7:rSluvbPEmiYAQP9k\Jhw8OMF@Wx8j0ff`j>o4nKW:Fxb[y;^^e<jh=d1fTaV9S3\Fzt3IuJ?ACp_QkdtuolPi?IE@qceQCjA85bFkp07Eu2;F1;c=PD3LqEkeGNga?fSYZ>=\DU7=raq@;kx\fDBU>sK?URWC@<ggDALMXNnpFpJOE9WrfLUQ6CQ2U3LN>psh]>zkujzgaPblPsYwC@s27;XTDb@9cFIP0p=GLPTBd9Xs2ti5nyJl_\zST0Jzv=acSxS0pifbD[P9Un1jfkrAuq5dF6qvqCHc=T0Qra27n6g6QkgQUpxw`PCG[ObqsTP6LghBy^@oDSz?4s\t:EhjpQAlKZ4Ohe4zI0?Ue0h1<8tk_tjgRF2YtiJkSXqi[QQn0EtEYSgYvSL9TT5sCPFqXfRX@7<;coVBCDIH>fe^A4ojpH1Y?UG1cD;5;2Eb__rMrqEo5bGQW`N3a0c?fNmH;ws=[F^[r@Y71CvsKuco88`:ViPqbBnk_>EAH`1ox6OaGrfM?kdr^lsJsW>L0h3<`Z4c0kb?RqUz>BC8l7F[nW_VsWpBk7Dw2e4pdz3UvYDKkVV>HwV<D[w8NDFefsSYS:<DL3ofx;QBCf5dV4;c[9QWTVMR4XMD3=EpOmwVa4=r1>3UlzDEc1m5>=q7@fSP4l0FowMIX@CWswGb3e^JEG>wZ9EdJkuns]N?FtcIfMOl;8wwTsZwbg8`aCwLk:;4hH4RBc
```

Reply:
Also, for the sake of completeness, these are the response headers:
```
± curl -I http://127.0.0.1:12345/test.php
HTTP/1.1 200 OK
Host: 127.0.0.1:12345
Date: Tue, 23 Jan 2024 16:17:58 GMT
Connection: close
X-Powered-By: PHP/8.2.1
Content-type: text/html; charset=UTF-8
```